packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -84,24 +82,34 @@ function SelectControl< ItemType = DefaultItemType >( {
 }: SelectControlProps< ItemType > ) {
 	const [ isFocused, setIsFocused ] = useState( false );
 	const [ inputValue, setInputValue ] = useState( '' );
-	const {
-		addSelectedItem,
-		getSelectedItemProps,
-		getDropdownProps,
-		removeSelectedItem,
-		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-		// @ts-ignore
-	} = useMultipleSelection( { itemToString: getItemLabel } );
 	let selectedItems = selected === null ? [] : selected;
 	selectedItems = Array.isArray( selectedItems )
 		? selectedItems
 		: [ selectedItems ].filter( Boolean );
+	const singleSelectedItem =
+		! multiple && selectedItems.length ? selectedItems[ 0 ] : null;
 	const filteredItems = getFilteredItems(
 		items,
 		inputValue,
 		selectedItems,
 		getItemLabel
 	);
+	const {
+		addSelectedItem,
+		getSelectedItemProps,
+		getDropdownProps,
+		removeSelectedItem,
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+	} = useMultipleSelection( { itemToString: getItemLabel, selectedItems } );
+
+	useEffect( () => {
+		if ( multiple ) {
+			return;
+		}
+
+		setInputValue( getItemLabel( singleSelectedItem ) );
```

Comment:
Do you think we should add an extra check around this if the `selectedItem` is not already equal to the `singleSelectedItem`? I always forget if React checks that internally or not, or will still trigger a re-render despite nothing changing.
Scratch that, looks like React won't trigger a re-render if the label is the same: https://dev.to/sunflower/reactjs-if-it-is-setting-a-state-with-the-same-value-will-the-component-be-re-rendered-5g24